dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ErrorCode.java

```diff
@@ -200,6 +200,7 @@
     /* ProgramRuleVariable validation */
     E4051( "A program rule variable with name `{0}` and program uid `{1}` already exists" ),
     E4052( "For program rule variable with name `{0}` following keywords are forbidden : and , or , not" ),
+    E4053( "ProgramStage `{0}` does not have reference to any Program." ),
```

Comment:
We could rephrase a bit to:

    Program stage `{0}` does not reference a program